frontend/src/metabase/parameters/components/Parameters.jsx

```diff
@@ -71,10 +71,14 @@ export default class Parameters extends Component {
         const queryParam = query && query[parameter.slug];
         if (queryParam != null || parameter.default != null) {
           let value = queryParam != null ? queryParam : parameter.default;
-          if (parameter.hasOnlyFieldTargets && value && !Array.isArray(value)) {
-            // FieldValuesWidget always produces an array. If this param has
-            // only field targets we'll use that widget, so we should start with
-            // an array to match.
+
+          // ParameterValueWidget uses FieldValuesWidget if there's no available
```

Comment:
>  Is it worth pushing this responsibility down into whatever component is rendered?

Yeah, it bothers me that `Parameters.jsx` has special logic relating to specific widgets. If it's not too difficult I think that would be preferable.